test/e2e/storage/drivers/csi_objects.go

```diff
@@ -117,3 +125,56 @@ func createGCESecrets(client clientset.Interface, config framework.VolumeTestCon
 	_, err = client.CoreV1().Secrets(config.Namespace).Create(s)
 	framework.ExpectNoError(err, "Failed to create Secret %v", s.GetName())
 }
+
+func waitForAllPodsInDaemonsetReady(f *framework.Framework, namespace string, dsName string) {
+	err := wait.PollImmediate(dsRetryPeriod, dsRetryTimeout, checkDaemonsetDesiredState(f, namespace, dsName))
+	framework.ExpectNoError(err, "daemonset %s/%s doesn't become ready: %v", namespace, dsName, err)
+}
+
+func checkDaemonsetDesiredState(f *framework.Framework, namespace string, dsName string) func() (bool, error) {
+	return func() (bool, error) {
+		ds, err := f.ClientSet.AppsV1().DaemonSets(namespace).Get(dsName, metav1.GetOptions{})
+		if err != nil {
+			return false, fmt.Errorf("Could not get daemonset %s/%s", namespace, dsName)
+		}
+		framework.Logf("daemonset %s/%s: desired:%d, ready:%d", namespace, dsName, ds.Status.DesiredNumberScheduled, ds.Status.NumberReady)
+		return ds.Status.DesiredNumberScheduled > 0 && ds.Status.DesiredNumberScheduled == ds.Status.NumberReady, nil
```

Comment:
I'm not sure if this will work for hostpath daemonset.  I believe we patch NodeName in the Daemonset for hostpath driver, which means that desired will == number of nodes, but ready will only ever be 1.

Reply:
As a fix, we could not patch NodeName into the hostpath daemonset.

Reply:
In my environment, desired for hostpath seems 1 even when there are multiple nodes. So, I hope that it should work as well in all environments.